client/express-checkout/blocks/components/express-checkout-preview.js

```diff
@@ -0,0 +1,107 @@
+/**
+ * External dependencies
+ */
+import { useState } from 'react';
+import { Elements, ExpressCheckoutElement } from '@stripe/react-stripe-js';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import InlineNotice from 'components/inline-notice';
+
+export const ExpressCheckoutPreviewComponent = ( {
+	stripe,
+	buttonType,
+	theme,
+	height,
+} ) => {
+	const [ canRenderButtons, setCanRenderButtons ] = useState( true );
+
+	const options = {
+		mode: 'payment',
+		amount: 1000,
+		currency: 'usd',
+	};
+
+	const mapThemeConfigToButtonTheme = ( paymentMethod, buttonTheme ) => {
+		switch ( buttonTheme ) {
+			case 'dark':
+				return 'black';
+			case 'light':
+				return 'white';
+			case 'light-outline':
+				if ( paymentMethod === 'googlePay' ) {
+					return 'white';
+				}
+
+				return 'white-outline';
+			default:
+				return 'black';
+		}
+	};
+
+	const googlePayType = buttonType === 'default' ? 'plain' : buttonType;
+
+	const applePayType = buttonType === 'default' ? 'plain' : buttonType;
+
+	let shouldOnlyShowApplePay = false;
+	if ( window.ApplePaySession && window.ApplePaySession.canMakePayments() ) {
+		shouldOnlyShowApplePay = true;
+	}
+
+	const buttonOptions = {
+		buttonHeight: Math.min( Math.max( height, 40 ), 55 ),
+		buttonTheme: {
+			googlePay: mapThemeConfigToButtonTheme( 'googlePay', theme ),
+			applePay: mapThemeConfigToButtonTheme( 'applePay', theme ),
+		},
+		buttonType: {
+			googlePay: googlePayType,
+			applePay: applePayType,
+		},
+		paymentMethods: {
+			link: 'never',
+			googlePay: shouldOnlyShowApplePay ? 'never' : 'always',
+			applePay: shouldOnlyShowApplePay ? 'always' : 'never',
+		},
+		layout: { maxColumns: 1, overflow: 'never' },
+	};
+
+	const onReady = ( { availablePaymentMethods } ) => {
+		if ( availablePaymentMethods ) {
+			setCanRenderButtons( true );
+		} else {
+			setCanRenderButtons( false );
+		}
+	};
+
+	if ( canRenderButtons ) {
+		return (
+			<div
+				key={ `${ buttonType }-${ height }-${ theme }` }
```

Comment:
I had to use the `key` prop here to force a rerender of the whole component when any of these props change because I found out that the `<ExpressCheckoutElement>` isn't updated when the `options` prop changes.

This causes an unfortunate "flash" where the button disappears entirely until it can be re-rendered with the new settings. But AFAICT there is no way around it.

_Although_ I guess we could render several ECE's, each with a different configuration of the settings and the settings just dictate which ones have `display: none` set? 🤔 
That probably wouldn't cause a flash like the current implementation, but it might make loading the page a little heavy? @pierorocca @elizaan36 @nikkivias what do we think here from a design perspective? (see video in description for a demo)

Reply:
Tested this a little more and was able to remove the `height` from the key. Also didn't need to include the border radius (now that #9010 is merged) so changing the height and border radius feels much better than the theme or CTA 🎉

Reply:
Here's a demonstration of the lag in re-rendering the buttons:


https://github.com/Automattic/woocommerce-payments/assets/13835680/c5d0b279-ab80-4648-9eeb-9fe3d99c7416

Reply:
The flash is a little odd, especially since only the Apple Pay and Google Pay buttons are flashing. 

> That probably wouldn't cause a flash like the current implementation, but it might make loading the page a little heavy?

Could you share more on how loading the page would be heavy?

Reply:
> Could you share more on how loading the page would be heavy?

Yes! The flashing is because the Stripe buttons are loaded in an `iframe` from Stripe's systems, and it takes a bit to initialize the buttons. The "solution" to the flashing I'm proposing would be to load all possible variations of the buttons, and hide them all via `display: none;`, except the one that matches the current configuration.

When the configuration is changed, instead of recreating the `iframe` with the buttons we would just hide the one that's currently visible and show the one that matches the selected configuration.

The page would be heavy to load because there would be 12 different `iframes` on the page, all showing different variations of the buttons (with only 1 of those visible at a time) instead of using just one `iframe`. Basically making the page slower to load, but it should be snappy once it's loaded.

> [!IMPORTANT]
> Important disclaimer: I haven't tested the proposed solution here, but I think it should work.

Reply:
@elizaan36 @pierorocca @nikkivias just to make sure this isn't holding up merging this PR I've created https://github.com/Automattic/woocommerce-payments/issues/9091 to track this issue specifically and we can keep working on it from there, if we decide this is something we need to improve.

Reply:
FYI @bborman22 ^